### ISSUE TITLE
Wrong constructor ownership when returning an anonymous class

### ISSUE THREAD

**Issue by DaGhostman** [infection/infection#682]
| Question    | Answer
| ------------| ---------------
| Infection version | 0.12.2
| Test Framework version | PHPUnit 7.5.8
| PHP version | 7.2.15-0ubuntu0.18.10.2 (NTS)
| Platform    | Ubuntu 18.10
| Github Repo | https://github.com/phOnion/framework/tree/feature/next

The issue I've encountered is when there is a class method, that returns an anonymous class, which has a constructor. The issue appears to be that it doesn't recognize which class owns the `__construct` method, this code example seems to reproduce the issue:

```php
class Foo {
    public function bar(): object
    {
        return new class() {
            public function __construct() { /* irrelevant */ }
        };
     }
}
```

I was able to pinpoint the issue to [src/Router/Strategy/Factory/RouteStrategyFactory](https://github.com/phOnion/framework/blob/feature/next/src/Router/Strategy/Factory/RouteStrategyFactory.php#L19). [This temporary workaround](https://gist.github.com/DaGhostman/a9c430adaf76001259658a65199c5707) seems to resolve the issue, however it feels nasty :smirk:

<details>
 <summary>Output with issue</summary>
 
 ```
In CodeCoverageMethodIgnoreVisitor.php line 56:
                                     
  Method __construct does not exist
 ```
</details>

**Comment by BackEndTea** [comment]
This looks like a duplicate of #616, which has been fixed with #630.  So this should be released with `0.13.0`. Could you check if dev-master works correctly for you?

I should've targeted the 0.12 branch with that fix, so it is my fault that its not in the `0.12` releases

**Comment by DaGhostman** [comment]
Yep, can confirm it is a duplicate, sorry for duplicating looking forward for the release
